assets/vue/components/MainLayout.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <div>
+    <span class="iaas__header-text">Identicons as a service</span>
+    <Identicon :src="src" :text="text" :size="size" />
+    <Form :text.sync="text" :size.sync="size" />
+  </div>
+</template>
+<script>
+import Form from "./Form.vue";
+import Identicon from "./Identicon.vue";
+export default {
+  name: "MainLayout",
+  components: {
+    Form,
+    Identicon,
+  },
+  data: () => ({
+    text: "",
+    size: 0,
+    src: "",
+  }),
+  created() {
+    const size = window[Symbol.for("size")];
+    const text = window[Symbol.for("text")];
+
+    this.size = size;
+    this.text = text;
+  },
+  methods: {
+    debounce(func, ms) {
+      if (this.timeout) clearTimeout(this.timeout);
+      this.timeout = setTimeout(func, ms);
+    },
+    setSrc() {
+      const encodedText = encodeURI(this.text);
+      this.debounce(
+        () => (this.src = `/${encodedText}/${this.size}/identicon.png`),
+        300
+      );
+    },
```

Comment:
We're not just setting the image `src` after 300ms of inactivity, but updating the `text` as well. Currently, the text updates before the image `src` updates.

Reply:
Thank you, nice catch!